packages/lib-utils/src/utils/url-sync.ts

```diff
@@ -0,0 +1,59 @@
+import type { History } from 'history';
+
+export const applyFiltersToURL = (
+  history: History,
+  keys: string[],
+  filterValues: Record<string, string>,
+) => {
+  const searchParams = new URLSearchParams(history.location.search);
+  keys.forEach((key: string) => {
+    searchParams.delete(key);
+    if (filterValues[key]) {
+      searchParams.set(key, filterValues[key]);
+    }
+  });
+
+  history.replace({
+    pathname: history.location.pathname,
+    search: searchParams.toString(),
+  });
+};
+
+export const syncFiltersWithURL = (
+  history: History,
+  keys: string[],
+  defaults: Record<string, string> = {},
+) => {
+  const searchParams = new URLSearchParams(history.location.search);
+
+  let filters: Record<string, string> = keys.reduce((acc, key) => {
+    const values = searchParams.getAll(key);
+    return {
+      ...acc,
+      [key]: values.length > 1 ? values : values[0],
+    };
+  }, {});
+
+  Object.keys(defaults).forEach((key: string) => {
+    const value = defaults[key];
+    filters = {
```

Comment:
@fhlavac is the only thing we're doing here is reassigning the value of `filters[key]` here to a default value if it's not set? We might want to favor chaining for readability - and we could also merge the objects together (defaults and parsed filters) using [lodash merge](https://lodash.com/docs/#merge). Something like: `_.merge(defaults, filters)`